app/scripts/lib/keyring-snaps-permissions.test.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  SubjectMetadataController,
+  SubjectType,
+} from '@metamask/permission-controller';
+import { KeyringRpcMethod } from '@metamask/keyring-api';
+import { keyringSnapPermissionsBuilder } from './keyring-snaps-permissions';
+
+describe('keyringSnapPermissionsBuilder', () => {
+  const mockController = {
+    getSubjectMetadata: jest.fn((origin: string) => {
+      if (origin.startsWith('https://') || origin.startsWith('http://')) {
```

Comment:
Just to make sure, is `http://` disallowed in stable?

Reply:
It's not filtered here, we just rely on the `SubjectMetadataController` to return the subject type.

I assume it would be better suited to be done in the Permissions Controller or when we check the `allowedOrigins`, but I can add a check here as a fail-safe, WDYT?

Reply:
If it's not enforced anywhere right now we should probably implement this in the caveat specification. I'm not sure how we would do that to support HTTP in Flask though. 🤔

Reply:
Oh I see this is implemented already.